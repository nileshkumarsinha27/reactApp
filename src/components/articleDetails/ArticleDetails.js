import React, { Component } from 'react';
import './style.css';
class ArticleDetails extends Component {
  state = {
    showMore: false
  };

  changeState = () =>
    this.setState(prevState => ({ showMore: !prevState.showMore }));

  generateArticleDOM = () => {
    const { title, url, byline, abstract, section } = this.props.data;
    return (
      <div>
        <h2>{title}</h2>
        <p>{byline}</p>
        {!this.state.showMore && (
          <span onClick={this.changeState}>Show More</span>
        )}
        {this.state.showMore && (
          <div>
            <p>Description: {abstract}</p>
            <p>
              Url:
              <a href={url} target="_blank" rel="noopener noreferrer">
                Click Here
              </a>
            </p>
            <p>Section: {section}</p>
            <span onClick={this.changeState}>Show Less</span>
          </div>
        )}
      </div>
    );
  };
  render(props, state) {
    return <div className="articledetails">{this.generateArticleDOM()}</div>;
  }
}
export default ArticleDetails;
