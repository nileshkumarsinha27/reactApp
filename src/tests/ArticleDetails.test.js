import { Provider } from 'react-redux';
import React from 'react';
import { createStore } from 'redux';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArticleDetails from '../components/articleDetails/ArticleDetails';
import rootReducers from '../rootReducer';
configure({ adapter: new Adapter() });
const store = createStore(rootReducers);

describe('Initial Test of ArticleDetails component', () => {
  test('it renders snapshot correctly', () => {
    const context = shallow(
      <Provider store={store}>
        <ArticleDetails />
      </Provider>
    );
    expect(context).toMatchSnapshot();
  });
});
