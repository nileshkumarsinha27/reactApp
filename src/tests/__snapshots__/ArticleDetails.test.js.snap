// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial Test of ArticleDetails component it renders snapshot correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ArticleDetails />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ArticleDetails />,
      "value": Object {
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeState": Object {
          "articles": Object {
            "articleList": Object {
              "count": 0,
              "list": Array [],
            },
          },
          "books": Object {
            "bookList": Object {
              "count": 0,
              "list": Array [],
            },
          },
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ArticleDetails />,
        "value": Object {
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "storeState": Object {
            "articles": Object {
              "articleList": Object {
                "count": 0,
                "list": Array [],
              },
            },
            "books": Object {
              "bookList": Object {
                "count": 0,
                "list": Array [],
              },
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
